Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -423,6 +428,14 @@ extension HTTPClient.Configuration {
             self.read = read
         }
     }
+
+    /// Specifies redirect processing settings.
+    public enum RedirectPolicy {
```

Comment:
I don't think we should have this as a `public enum` because that means we can never add policies. I would recommend making this a private enum and wrapping it into a `public struct RedirectPolicy` with all the enum cases as

```
public struct RedirectPolicy {
    private enum Policy {
        case disallow
        ...
    }
    private var policy: Policy

    public static let disallow = RedirectPolicy(policy: .disallow)

    public static func allow(max: Int, allowCycles: Bool) -> RedirectPolicy {
        return RedirectPolicy(policy: .allow(max, allowCycles)
    }
}
```

Reply:
Agreed :D, fixed, thanks!